### PR TITLE
Don't generate functional types for functions with long signatures

### PR DESCRIPTION
**Pull request by ilmat192**
The K/N stdlib doesn't contain basic classes (e.g. Functon0,
Function1 etc) for functional types with >22 parameters.
So we cannot convert functional pointers with such signatures
into Kotlin functional types and have to replace them with
opaque pointers.

Fix for GitHub issue #2334.

### CODE REVIEW

File: Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/Indexer.kt [hunk 1]
```diff
@@ -618,6 +618,13 @@ internal class NativeIndexImpl(val library: NativeLibrary) : NativeIndex() {
         } else {
             val returnType = convertType(clang_getResultType(type))
             val numArgs = clang_getNumArgTypes(type)
+
+            // Ignore functions with long signatures since we have no basic class for such functional types in the stdlib.
+            // TODO: Remove this limitation when functional types with long signatures are supported.
+            if (numArgs > 22) {
```
Comment (sbogolepov) [inline Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/Indexer.kt]: Maybe add some kind of warning?

File: backend.native/tests/interop/basics/funptr.kt [hunk 2]
```diff
@@ -31,6 +36,11 @@ fun main(args: Array<String>) {
     printIntPtr(isIntPositivePtr(-42).ifThenOneElseZero())
 
     assertEquals(getMaxUIntGetter()!!(), UInt.MAX_VALUE)
+
+    val longSignaturePtr: COpaquePointer? = getLongSignatureFunctionPtr()
```
[the remark this answers is not in the document]
Comment (ilmat192) [reply backend.native/tests/interop/basics/funptr.kt]: It's impossible to get a `KClass` of an interop class instance: the `getPointeeType` function returns something named `unreflected class` in this case.

I don't see other ways to check class of this variable at runtime. @SvyatoslavScherbina WDYT, is it ok to merge this PR without changing the test?

File: Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/Indexer.kt (hunk 1 above)
Comment (ilmat192) [reply Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/Indexer.kt]: Fixed.

File: backend.native/tests/interop/basics/funptr.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/interop/basics/funptr.kt]: > I don't see other ways to check class of this variable at runtime.

There are a lot of tricks to check class at compile time :)
But it doesn't make much sense to check this actually, so let's ignore this for now.

> is it ok to merge this PR without changing the test?

Yes, it is ok, I have already approved this PR.

[the remark this answers is not in the document]
Comment (ilmat192) [reply backend.native/tests/interop/basics/funptr.kt]: Ok, I'll wait for windows and mac tests and merge the PR.

**Pull request merged**